packages/perseus-core/src/widgets/matcher/matcher-util.test.ts

```diff
@@ -0,0 +1,54 @@
+import getMatcherPublicWidgetOptions from "./matcher-util";
+
+import type {PerseusMatcherWidgetOptions} from "../../data-schema";
+
+describe("getMatcherPublicWidgetOptions", () => {
+    it("should return shuffled right array when order doesn't matter", () => {
+        // Arrange
+        const options: PerseusMatcherWidgetOptions = {
+            labels: ["**Claims**", "**Evidence**"],
+            padding: true,
+            orderMatters: false,
+            right: ["Stars", "Earth", "Life", "Rapid", "Milky Way"],
+            left: ["Fuel", "Plate", "Hydrogen", "Average", "Billion"],
+        };
+
+        // Act
+        const publicWidgetOptions = getMatcherPublicWidgetOptions(options);
+
+        // Assert
+        expect(publicWidgetOptions).toEqual({
+            labels: ["**Claims**", "**Evidence**"],
+            padding: true,
+            orderMatters: false,
+            right: ["Milky Way", "Stars", "Life", "Earth", "Rapid"],
+            left: ["Fuel", "Plate", "Hydrogen", "Average", "Billion"],
+        });
+        expect(publicWidgetOptions.right).not.toEqual(options.right);
+    });
+
+    it("should return shuffled left and right arrays when order matters", () => {
+        // Arrange
+        const options: PerseusMatcherWidgetOptions = {
+            labels: ["**Claims**", "**Evidence**"],
+            padding: true,
+            orderMatters: true,
+            right: ["Stars", "Earth", "Life", "Rapid", "Milky Way"],
+            left: ["Fuel", "Plate", "Hydrogen", "Average", "Billion"],
+        };
+
+        // Act
+        const publicWidgetOptions = getMatcherPublicWidgetOptions(options);
+
+        // Assert
+        expect(publicWidgetOptions).toEqual({
+            labels: ["**Claims**", "**Evidence**"],
+            padding: true,
+            orderMatters: true,
+            right: ["Milky Way", "Stars", "Life", "Earth", "Rapid"],
+            left: ["Billion", "Fuel", "Hydrogen", "Plate", "Average"],
+        });
+        expect(publicWidgetOptions.right).not.toEqual(options.right);
+        expect(publicWidgetOptions.left).not.toEqual(options.left);
```

Comment:
I added these expect lines to emphasize what we are looking for in this test. I think these check what we need to check regarding making sure the public widget options don't have the same order as the original ones, but please let me know if there's another test I can add for better coverage!